packages/wonder-blocks-icon-button/src/components/icon-button-core.tsx

```diff
@@ -64,20 +69,25 @@ const IconButtonCore: React.ForwardRefExoticComponent<
                 ? SemanticColor.controlDestructive
                 : SemanticColor.controlDefault;
 
-        const buttonStyles = _generateStyles(buttonColor, kind, light);
+        const buttonStyles = _generateStyles(buttonColor, kind, light, size);
 
         const defaultStyle = [
             sharedStyles.shared,
-            disabled && sharedStyles.disabled,
             buttonStyles.default,
-            disabled && buttonStyles.disabled,
+            disabled && [sharedStyles.disabled, buttonStyles.disabled],
```

Comment:
**nit:** By looking at the code, it looks like `sharedStyles.disabled` can be removed as both styles share the `cursor` key.

BTW thanks for refactoring this! It made me realize about that duplication 👏